packages/mds-compliance-engine/engine/helpers.ts

```diff
@@ -78,20 +148,15 @@ export function isRuleActive(rule: Rule<Exclude<RULE_TYPE, 'rate'>>): boolean {
     throw new RuntimeError(`TIMEZONE environment variable ${env.TIMEZONE} is not a valid timezone!`)
   }
 
-  const local_time = moment().tz(env.TIMEZONE)
-
-  if (!rule.days || rule.days.includes(Object.values(DAYS_OF_WEEK)[local_time.day()] as DAY_OF_WEEK)) {
-    if (!rule.start_time || local_time.isAfter(moment(rule.start_time, TIME_FORMAT))) {
-      if (!rule.end_time || local_time.isBefore(moment(rule.end_time, TIME_FORMAT))) {
-        return true
-      }
-    }
-  }
-  return false
+  return isCurrentDayInDays({ days }) && isCurrentTimeInInterval({ start_time, end_time })
 }
 
 export function isInVehicleTypes(rule: Rule<Exclude<RULE_TYPE, 'rate'>>, device: Device): boolean {
-  return !rule.vehicle_types || (rule.vehicle_types && rule.vehicle_types.includes(device.vehicle_type))
+  return (
+    !rule.vehicle_types ||
+    rule.vehicle_types.length === 0 ||
+    (rule.vehicle_types && rule.vehicle_types.includes(device.vehicle_type))
```

Comment:
why are we still checking rule.vehicle_types is truthy here?

Reply:
Because the type of `vehicle_types` here is `?: Nullable<VEHICLE_TYPE[]>`, so it could be one of:
```
undefined | null | Array<VEHICLE_TYPE>
```

the `undefined` bit comes from the `?:` operator in the original typedef for this property, which can be found in packages/mds-policy-service/@types/models.ts#83